setuptools/tests/test_sdist.py

```diff
@@ -92,9 +92,10 @@ def latin1_fail():
 class TestSdistTest:
     def setup_method(self, method):
         self.temp_dir = tempfile.mkdtemp()
-        f = open(os.path.join(self.temp_dir, 'setup.py'), 'w')
-        f.write(SETUP_PY)
-        f.close()
+        if not method == self.test_setup_py_missing:
```

Comment:
I don't love this, it unnecessarily couples `setup_method` and `test_setup_py_missing`.

I'd rather just have `test_setup_py_missing` in a different class, or create a more general method for registering methods that don't need this setup.